src/java.base/share/classes/java/net/URLClassLoader.java

```diff
@@ -1086,13 +1086,19 @@ public ClassFinder(String name, ClassLoader loader) {                       //Op
      }                                                                           //OpenJ9-shared_classes_misc
                                                                                  //OpenJ9-shared_classes_misc
      public Class<?> run() throws ClassNotFoundException {                         //OpenJ9-shared_classes_misc
-	String path = name.replace('.', '/').concat(".class");                   //OpenJ9-shared_classes_misc
+        String path = name.replace('.', '/').concat(".class");                   //OpenJ9-shared_classes_misc
+        Resource res;
```

Comment:
This doesn't compile:
```
src/java.base/share/classes/java/net/URLClassLoader.java:1098: error: variable res might not have been initialized
            if (res.getDataError() != null) {                                    //OpenJ9-shared_classes_misc
                ^
```
I suggest this method should follow the flow of the upstream code:
```
    public Class<?> run() throws ClassNotFoundException {                       //OpenJ9-shared_classes_misc
        String path = name.replace('.', '/').concat(".class");                  //OpenJ9-shared_classes_misc
        Resource res = ucp.getResource(path, false);                            //OpenJ9-shared_classes_misc
        if (res != null) {                                                      //OpenJ9-shared_classes_misc
            try {                                                               //OpenJ9-shared_classes_misc
                return defineClass(name, res);                                  //OpenJ9-shared_classes_misc
            } catch (IOException e) {                                           //OpenJ9-shared_classes_misc
                throw new ClassNotFoundException(name, e);                      //OpenJ9-shared_classes_misc
            } catch (ClassFormatError e2) {                                     //OpenJ9-shared_classes_misc
                if (res.getDataError() != null) {                               //OpenJ9-shared_classes_misc
                    e2.addSuppressed(res.getDataError());                       //OpenJ9-shared_classes_misc
                }                                                               //OpenJ9-shared_classes_misc
                throw e2;                                                       //OpenJ9-shared_classes_misc
            }                                                                   //OpenJ9-shared_classes_misc
        } else {                                                                //OpenJ9-shared_classes_misc
            return null;                                                        //OpenJ9-shared_classes_misc
        }                                                                       //OpenJ9-shared_classes_misc
    }                                                                           //OpenJ9-shared_classes_misc
```

Reply:
It appeared in my local build as well.
Note: `Resource res = ucp.getResource(path, false)` lost shared class patch.